src/include/module/module/generic.h

```diff
@@ -0,0 +1,138 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *	   Adrian Warecki <adrian.warecki@intel.com>
+ */
+
+#ifndef __MODULE_MODULE_GENERIC__
+#define __MODULE_MODULE_GENERIC__
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#include "interface.h"
+#include "../ipc4/base-config.h"
```

Comment:
```<module/ipc4/base-config.h>``` ?

Reply:
This file is used by both sof and a loadable module. From the sof perspective it is accessible by `<module/ipc4/base-config.h>`, and from the loadable module perspective it is `<ipc4/base-config.h>`. That's why I used a relative path to handle both cases.

Reply:
We should set this correctly in the build for baseFW and for modules. i.e. we use ```<module/ipc4/base-config.h>``` and use the ```-I``` directive depending on build target

Reply:
@lgirdwood If I specify the path to `src/include` instead of `src/include/module` via `-I` when building a loadable module, the modules will be able to include any sof header file, which is what we want to avoid. Should I move the `module` directory to another place that will be added to the include directories by both sof and modules?

Reply:
Modules today can and do include any header file (we still have a lot of header refactoring work todo) and we have vice versa where non module code (testbench, UTs, plugin) will need to include module headers.